install.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Fancy colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color aka reset
+
+# Version to install. Defaults to latest or set by --version or -v
+VERSION=""
+
+# Print in colors - 0=green, 1=red, 2=neutral
+# e.g. fancy_print 0 "All is great"
+fancy_print() {
+    if [[ $1 == 0 ]]; then
+        echo -e "${GREEN}${2}${NC}"
+    elif [[ $1 == 1 ]]; then
+        echo -e "${RED}${2}${NC}"
+    else
+        echo -e "${2}"
+    fi
+}
+
+# Function to print the help message
+print_help() {
+    fancy_print 2 ""
+    fancy_print 2 "---- Spin Installer Script ----"
+    fancy_print 2 "This script installs Spin in the current directory."
+    fancy_print 2 ""
+    fancy_print 2 "Comand line arguments"
+    fancy_print 2 "--version or -v  : Provide what version to install e.g. \"v0.5.0\" or \"canary\"."
+    fancy_print 2 "--help    or -h  : Shows this help message"
+}
+
+# Function used to check if utilities are available
+require() {
+    if ! hash "$1" &>/dev/null; then
+        fancy_print 1 "'$1' not found in PATH. This is required for this script to work."
+        exit 1
+    fi
+}
+
+# Parse input arguments
+while [[ $# -gt 0 ]]; do
+    case $1 in
+    '--version' | -v)
+        shift
+        if [[ $# -ne 0 ]]; then
+            VERSION="${1}"
+        else
+            fancy_print 1 "Please provide the desired version. e.g. --version v0.5.0 or -v canary"
+            exit 0
+        fi
+        ;;
+    '--help' | -h)
+        shift
+        print_help
+        ;;
+    *)
+        fancy_print 1 "Unknown argument ${1}."
+        print_help
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+# Check all required utilities are available
+require wget
+require tar
+require uname
+
+# Check if we're on a suppoerted system and get OS and processor architecture to download the right version
+UNAME_ARC=$(uname -m)
+
+case $UNAME_ARC in
+"x86_64")
+    ARC="amd64"
+    ;;
+"arm64")
+    ARC="aarch64"
+    ;;
+*)
+    fancy_print 1 "The Processor: ${UNAME_ARC} is not supported by Spin."
+    exit 1
+    ;;
+esac
+
+case $OSTYPE in
+"linux-gnu"*)
+    OS="linux"
```

Comment:
Do we want to handle the specific case when OSTYPE is linux-gnu and ARC is aarch64 here?  eg print a line stating this combination isn't yet supported and exit 1?

Reply:
Oh - I see the combination of Linux and aarch64 is not caught as not-supported.

Reply:
uname -m on Mac return "arm64" on Linux it returns "aarch64", so this was originally caught when evaluating the processor type, but now I've made that case specific.